specs/commandline/sign.md

```diff
@@ -73,20 +75,24 @@ Notation uses empty JSON object `{}` as the default configuration content, and t
 # - A signing plugin is installed. See plugin documentation (https://github.com/notaryproject/notaryproject/blob/main/specs/plugin-extensibility.md) for more details.
 # - Configure the signing plugin as instructed by plugin vendor.
 
-# Add a default signing key referencing the remote key identifier, and the plugin associated with it.
-notation key add --default --name <key_name> --plugin <plugin_name> --id <remote_key_id>
-
-# sign an artifact stored in a registry
-notation sign <registry>/<repository>@<digest>
+notation sign --plugin <plugin_name> --id <remote_key_id> <registry>/<repository>@<digest>
 ```
-
 An example for a successful signing:
 
 ```console
-$ notation sign localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
+$ notation sign --plugin <plugin_name> --id <remote_key_id> localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
 Successfully signed localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
 ```
 
+### Sign an OCI artifact by adding new key
+```shell
+# Add a default signing key referencing the remote key identifier, and the plugin associated with it.
+notation key add --default --name <key_name> --plugin <plugin_name> --id <remote_key_id>
+
+# sign an artifact stored in a registry
+notation sign <registry>/<repository>@<digest>
+```
```

Comment:
This method should be the suggested way to do signing.

We should call `notation sign --plugin <plugin_name> --id <remote_key_id>` as something like "sign an OCI artifact with on-demand remote key" and make it less perferrable than "Sign an OCI artifact by adding new key".

Otherwise, users will do
```sh
notation sign --plugin <plugin_name> --id <remote_key_id> localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
notation sign --plugin <plugin_name> --id <remote_key_id> localhost:5000/net-monitor@sha256:2d5154312d3727d3d6f3c548ccd96158b001bc713a573cf5569941e0f5c331c8
notation sign --plugin <plugin_name> --id <remote_key_id> localhost:5000/net-monitor@sha256:2033f049a95d2227a5d16b16bc0ef197870b90f1bf95d452531f15c902c4e9bf
notation sign --plugin <plugin_name> --id <remote_key_id> localhost:5000/net-monitor@sha256:e6c2ade679277cb93dacf8b3cb23d05c6368dbfeefd212d3a9b7f7b67e67bc58
```
instead of
```sh
notation key add --default --name <key_name> --plugin <plugin_name> --id <remote_key_id>
notation sign localhost:5000/net-monitor@sha256:b94d27b9934d3e08a52e52d7da7dabfac484efe37a5380ee9088f7ace2efcde9
notation sign localhost:5000/net-monitor@sha256:2d5154312d3727d3d6f3c548ccd96158b001bc713a573cf5569941e0f5c331c8
notation sign localhost:5000/net-monitor@sha256:2033f049a95d2227a5d16b16bc0ef197870b90f1bf95d452531f15c902c4e9bf
notation sign localhost:5000/net-monitor@sha256:e6c2ade679277cb93dacf8b3cb23d05c6368dbfeefd212d3a9b7f7b67e67bc58
```

Reply:
Yeah that sounds reasonable, will update PR to make `notation key add --default --name <key_name> --plugin <plugin_name> --id <remote_key_id>` as preferable method.

Reply:
+1 To @shizhMSFT.